Remove variable length arrays in structs (VLAIS) from HMAC

The use of VLAIS in the Linux Kernel code precludes the use of compilers which
don't implement VLAIS (for instance the Clang compiler). Since ctx is always
32-bit, hard coding a size of 4 accomplishes the same thing without the use of
VLAIS.

Signed-off-by: Mark Charlebois <charlebm@gmail.com>
Reviewed-by: Behan Webster <behanw@converseincode.com>
---
 hmac.c    |    2 +-
 testmgr.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: linux/crypto/testmgr.c
===================================================================
--- linux.orig/crypto/testmgr.c
+++ linux/crypto/testmgr.c
@@ -1478,7 +1478,7 @@
 	do {
 		struct {
 			struct shash_desc shash;
-			char ctx[crypto_shash_descsize(tfm)];
+			char ctx[4];
 		} sdesc;
 
 		sdesc.shash.tfm = tfm;
Index: linux/crypto/hmac.c
===================================================================
--- linux.orig/crypto/hmac.c
+++ linux/crypto/hmac.c
@@ -54,7 +54,7 @@
 	struct crypto_shash *hash = ctx->hash;
 	struct {
 		struct shash_desc shash;
-		char ctx[crypto_shash_descsize(hash)];
+		char ctx[4];
 	} desc;
 	unsigned int i;
 
